PublicGitArchive/pga/pga/pga.go

```diff
@@ -3,202 +3,46 @@
 package pga
 
 import (
-	"encoding/csv"
 	"fmt"
 	"io"
 	"strconv"
 	"strings"
 )
 
-// Repository contains the data used to filter the index and download files.
-type Repository struct {
-	URL       string   `json:"url"`           // URL of the repository.
-	Filenames []string `json:"sivaFilenames"` // Siva filenames.
-	Size      int64    `json:"size"`          // Sum of the siva files sizes.
-	License   string   `json:"license"`       // Main license name of the repository.
-
-	// Stats per language
-	Languages             []string `json:"langs"`             // Languages found in the repository.
-	LanguagesByteCount    []int64  `json:"langsByteCount"`    // Number of bytes for the language in same index.
-	LanguagesLineCount    []int64  `json:"langsLinesCount"`   // Number of lines of code for the language in same index.
-	LanguagesFileCount    []int64  `json:"langsFilesCount"`   // Number of files in the language in the same index.
-	LanguagesEmptyLines   []int64  `json:"emptyLinesCount"`   // Number of blank lines for the language in the same index
-	LanguagesCodeLines    []int64  `json:"codeLinesCount"`    // Number of lines of code for the language in the same index.
-	LanguagesCommentLines []int64  `json:"commentLinesCount"` // Number of comment lines for the language in the same index.
-
-	// Global stats
-	Files    int64 `json:"fileCount"`     // Number of files in the repository.
-	Commits  int64 `json:"commitsCount"`  // Number of commits in the repository.
-	Branches int64 `json:"branchesCount"` // Number of branches in the repository.
-	Forks    int64 `json:"forkCount"`     // Number of forks of this repository.
-	Stars    int64 `json:"stars"`         // Number of stars of this repository.
+// Repository provides abstraction for getters and conversion to CSV format.
+type Repository interface {
+	ToCSV() []string
+	GetURL() string
+	GetLanguages() []string
+	GetFilenames() []string
 }
 
-// CSVHeaders are the headers expected for the CSV formatted index.
-func CSVHeaders() []string {
-	cp := make([]string, len(csvHeaders))
-	copy(cp, csvHeaders)
-	return cp
+// Dataset provides abstraction for creating an Index from a CSV file.
+type Dataset interface {
+	IndexFromCSV(r io.Reader) (Index, error)
```

Comment:
Add `Name() string` and refactor `handleDatasetArg`

Reply:
done